src/main/java/bio/terra/service/dataset/flight/datadelete/DataDeletionCopyFilesToBigQueryScratchBucketStep.java

```diff
@@ -0,0 +1,64 @@
+package bio.terra.service.dataset.flight.datadelete;
+
+import static bio.terra.service.dataset.flight.datadelete.DataDeletionUtils.getDataset;
+import static bio.terra.service.dataset.flight.datadelete.DataDeletionUtils.getRequest;
+
+import bio.terra.common.FlightUtils;
+import bio.terra.model.DataDeletionRequest;
+import bio.terra.model.DataDeletionTableModel;
+import bio.terra.service.common.gcs.CommonFlightKeys;
+import bio.terra.service.common.gcs.GcsUriUtils;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.dataset.flight.ingest.SkippableStep;
+import bio.terra.service.filedata.google.gcs.GcsPdao;
+import bio.terra.service.resourcemanagement.google.GoogleBucketResource;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import com.google.cloud.storage.BlobId;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+
+public class DataDeletionCopyFilesToBigQueryScratchBucketStep extends SkippableStep {
+
+  private final DatasetService datasetService;
+  private final GcsPdao gcsPdao;
+
+  public DataDeletionCopyFilesToBigQueryScratchBucketStep(
+      DatasetService datasetService, GcsPdao gcsPdao, Predicate<FlightContext> skipCondition) {
+    super(skipCondition);
+    this.datasetService = datasetService;
+    this.gcsPdao = gcsPdao;
+  }
+
+  @Override
+  public StepResult doSkippableStep(FlightContext context)
+      throws InterruptedException, RetryException {
+    FlightMap workingMap = context.getWorkingMap();
+    Dataset dataset = getDataset(context, datasetService);
+    String projectId = dataset.getProjectResource().getGoogleProjectId();
+    DataDeletionRequest dataDeletionRequest = getRequest(context);
+    Set<String> tablesNeedingCopy =
+        FlightUtils.getTyped(workingMap, DataDeletionMapKeys.TABLE_NAMES_NEEDING_COPY);
+    GoogleBucketResource bucketResource =
+        FlightUtils.getTyped(workingMap, CommonFlightKeys.SCRATCH_BUCKET_INFO);
+    List<DataDeletionTableModel> tables = dataDeletionRequest.getTables();
+    for (var table : tables) {
+      if (tablesNeedingCopy.contains(table.getTableName())) {
+        BlobId from = GcsUriUtils.parseBlobUri(table.getGcsFileSpec().getPath());
+        String to =
+            GcsUriUtils.getGsPathFromComponents(
+                bucketResource.getName(),
+                String.format("%s/%s", context.getFlightId(), from.getName()));
+        gcsPdao.copyGcsFile(from, GcsUriUtils.parseBlobUri(to), projectId);
```

Comment:
This use of `GcsUriUtils.getGsPathFromComponents()` seems unnecessary here. If `copyGcsFile()` needs a `BlobId`, it would be simpler to make the `BlobId` directly instead of first making a path and then parsing the path.
```suggestion
        BlobId to =
            BlobId.of(
                bucketResource.getName(),
                String.format("%s/%s", context.getFlightId(), from.getName()));
        gcsPdao.copyGcsFile(from, to, projectId);
```

I guess the downside is that you now need to convert a `BlobId` to a GCS path, which I don't see a way to do using the `BlobId` APIs. But maybe adding `GcsUriUtils.toPath(BlobId)` would be better than what the code is currently doing.

Reply:
I've updated this so that both `from` and `to` are `BlobId`s now. Still needs to be converted to a `String` when passed to the flight map, since I'm pretty sure we prefer scalar values over objects (at least, that's the justification I've heard for passing `datasetId` around everywhere instead of the dataset object itself). The path creation has been pushed into `GcsUriUtils`, since its repeated code.